presto-product-tests/src/main/java/io/prestosql/tests/hive/TestViewfsHiveTable.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.tests.hive;
+
+import com.google.common.io.Resources;
+import com.google.inject.Inject;
+import io.prestosql.tempto.AfterTestWithContext;
+import io.prestosql.tempto.BeforeTestWithContext;
+import io.prestosql.tempto.ProductTest;
+import io.prestosql.tempto.hadoop.hdfs.HdfsClient;
+import org.testng.annotations.Test;
+
+import java.io.InputStream;
+
+import static io.prestosql.tempto.assertions.QueryAssert.Row.row;
+import static io.prestosql.tempto.assertions.QueryAssert.assertThat;
+import static io.prestosql.tempto.query.QueryExecutor.query;
+import static io.prestosql.tests.TestGroups.VIEWFS;
+import static io.prestosql.tests.utils.QueryExecutors.onHive;
+
+public class TestViewfsHiveTable
+        extends ProductTest
+{
+    @Inject
+    private HdfsClient hdfsClient;
+
+    @BeforeTestWithContext
+    public void setup()
+            throws Exception
+    {
+        hdfsClient.createDirectory("/user/hive/warehouse/TestViewfsTable/single_column");
+        try (InputStream inputStream = Resources.asByteSource(Resources.getResource("textfile/single_column.textfile")).openStream()) {
+            hdfsClient.saveFile("/user/hive/warehouse/TestViewfsTable/single_column/single_column.textfile", inputStream);
+        }
+    }
+
+    @AfterTestWithContext
+    public void cleanup()
+    {
+        hdfsClient.delete("/user/hive/warehouse/TestViewfsTable");
+    }
+
+    @Test(groups = VIEWFS)
+    public void testSelectViewfsTable()
+    {
+        onHive().executeQuery("DROP TABLE IF EXISTS test_select_viewfs");
+        onHive().executeQuery("" +
+                "CREATE TABLE test_select_viewfs" +
+                " (name string) " +
+                "STORED AS TEXTFILE " +
+                "LOCATION 'viewfs://hadoop-viewfs/default/TestViewfsTable/single_column'");
```

Comment:
Can we create this table in Presto, with `external_location`?

Reply:
Unfortunately, we can't create a table for viewfs on Presto. This is a known issue and even latest version also fails https://github.com/prestosql/presto/issues/1305.

Reply:
`Failed checking path` is thrown in 4 places. 
if you have full stack trace, can you add it in https://github.com/prestosql/presto/issues/1305 for reference?

Reply:
Oh sorry, it was my misconfiguration. We can create an external table.